cli/update_test.go

```diff
@@ -333,18 +335,16 @@ func TestUpdateValidateRichParameters(t *testing.T) {
 			assert.NoError(t, err)
 		}()
 
-		matches := []string{
-			stringParameterName, "$$",
-			"does not match", "",
-			"Enter a value", "abc",
-		}
-		for i := 0; i < len(matches); i += 2 {
-			match := matches[i]
-			value := matches[i+1]
-			pty.ExpectMatch(match)
-			pty.WriteLine(value)
-		}
-		<-doneChan
+		pty.ExpectMatch(stringParameterName)
+		pty.ExpectMatch("> Enter a value (default: \"\"): ")
```

Comment:
I understand the rationale behind proposing this matcher. However, I'm a bit surprised to observe that stdin isn't buffered when input arrives prematurely. Is this the intended behavior, or are you exploring different matcher configurations to address the issue?

Reply:
It's a good point.  I looked into this more and there is a race in `cliui.Prompt` that causes input to be lost in cases where input is being sent too early and we fail validation.

https://github.com/coder/internal/issues/203

I'll separately work on a fix for this ^^

I still think this PR is an improvement: in many cases the explicit matches are fewer lines than the looping construct, and are closer to the timing a human would have.

It also prevents these tests from deadlocking on failure via the bare `<-doneChan` read.

Reply:
I agree, mitigating the risk of deadlocks is certainly valuable here.

As for the explicit matchers, I also find them preferable from a readability standpoint; they make the intent clear and reduce ambiguity, allowing developers to quickly grasp the expected input without needing to infer it.